docs/src/_guide/configuration.md

```diff
@@ -52,6 +52,50 @@ can be used to pass along arbitrary key-value data to the new process.
 Meltano [reads settings from environment variables](#configuring-settings) when you run the [`meltano` command](/reference/command-line-interface),
 and populates them when it [evaluates plugin configuration](#expansion-in-setting-values)
 and [invokes plugin executables](#accessing-from-plugins).
+Meltano also supports specifying environment variables under the `env:` keys of `meltano.yml`, a Meltano Environment, or on the Plugin.
+
+### Specifying environment variables
+
+In addition to the terminal environment and the `.env ` file, Meltano supports the specification of environment variables at the following configuration levels:
+
+```yaml
+env:
+  # root level env
+  MY_ENV_VAR: top_level_env_var
+plugins:
+  extractors:
+    - name: tap-google-analytics
+      variant: meltano
+      env:
+        # root level plugin env
+        MY_ENV_VAR: plugin_level_env_var
+environments:
+  - name: dev
+    env:
+      # environment level env
+      MY_ENV_VAR: environment_level_env_var
+    config:
+      plugins:
+        extractors:
+          - name: tap-google-analytics
+            variant: meltano
+            env:
+              # environment level plugin env
+              MY_ENV_VAR: environment_level_plugin_env_var
+```
+
+Environment levels within `meltano.yml` resolve in order of precedence (within a plugins context), as you would expect:
```

Comment:
I think we can drop the `as you would expect` here. For newer folks I don't know what the clear expectation actually is. 

@kgpayne what do you think of adding integration tests like what Florian recently added to this? Looking at the docs I think I understand what the expectation is, but it'd be nice to have a few examples with specific integration tests so we can be very confident and clear about what the actual behavior is.

Reply:
Dropped the `as you would expect` 👍

It would definitely be useful to add integration tests, though there are two issues I think we should tackle first:
- https://github.com/meltano/meltano/issues/6326
- https://github.com/meltano/meltano/issues/6363

This would allow us to i) target all levels of the supported hierarchy from the CLI and ii) validate that resolution is actually working as expected (by printing the resolved environment at various points as higher-precedence env vars are added).

This issue has also been going for a few weeks already too, so I'd prefer not to expand scope if integration tests aren't a hard ask 🙂